django/utils/safestring.py

```diff
@@ -35,10 +35,12 @@ def __add__(self, rhs):
         Concatenating a safe string with another safe bytestring or
         safe string is safe. Otherwise, the result is no longer safe.
         """
-        t = super().__add__(rhs)
         if isinstance(rhs, SafeData):
-            return SafeString(t)
-        return t
+            return SafeString(super().__add__(rhs))
+        if isinstance(rhs, str):
+            return super().__add__(rhs)
```

Comment:
I think a slightly more clear layout for this logic could be:

```suggestion
        if isinstance(rhs, str):
            t = super().__add__(rhs)
            if isinstance(rhs, SafeData):
                t =  SafeString(t)
            return t
```